robot_calibration/src/optimizer.cpp

```diff
@@ -146,32 +144,42 @@ int Optimizer::optimize(OptimizationParams& params,
 
         // Create the block
         ceres::CostFunction * cost = Camera3dToArmError::Create(
-          dynamic_cast<Camera3dModel*>(models_[camera_name]),
-          models_[arm_name],
-          offsets_.get(), data[i]);
+            dynamic_cast<Camera3dModel*>(models_[camera_name]),
+            models_[arm_name],
+            offsets_.get(), data[i]);
+
+        int index = -1;
+        for (size_t k =0; k < data[i].observations.size() ; k++)
+        {
+            if ( data[i].observations[k].sensor_name == camera_name)
+            {
+             index = k;
+             break;
+            }
+        }
 
         if (progress_to_stdout)
         {
           double ** params = new double*[1];
           params[0] = free_params;
-          double * residuals = new double[data[i].observations[0].features.size() * 3];  // TODO: should check that all features are same length?
+          double * residuals = new double[data[i].observations[index].features.size() * 3];  // TODO: should check that all features are same length?
 
           cost->Evaluate(params, residuals, NULL);
           std::cout << "INITIAL COST (" << i << ")" << std::endl << "  x: ";
-          for (size_t k = 0; k < data[i].observations[0].features.size(); ++k)
+          for (size_t k = 0; k < data[i].observations[index].features.size(); ++k)
             std::cout << "  " << std::setw(10) << std::fixed << residuals[(3*k + 0)];
           std::cout << std::endl << "  y: ";
-          for (size_t k = 0; k < data[i].observations[0].features.size(); ++k)
+          for (size_t k = 0; k < data[i].observations[index].features.size(); ++k)
             std::cout << "  " << std::setw(10) << std::fixed << residuals[(3*k + 1)];
           std::cout << std::endl << "  z: ";
-          for (size_t k = 0; k < data[i].observations[0].features.size(); ++k)
+          for (size_t k = 0; k < data[i].observations[index].features.size(); ++k)
             std::cout << "  " << std::setw(10) << std::fixed << residuals[(3*k + 2)];
           std::cout << std::endl << std::endl;
         }
 
         problem->AddResidualBlock(cost,
-                                   NULL /* squared loss */,
-                                   free_params);
+                                  NULL,//squared loss ,
```

Comment:
need 2 spaces after , and one space after // (and remove extra comma end of line)